python_modules/dagster/dagster_tests/general_tests/parent_asset_graph_differ_tests/test_parent_asset_graph_diff.py

```diff
@@ -119,23 +119,135 @@ def test_new_asset(instance):
+        code_location_to_diff="basic_asset_graph",
+        parent_code_locations=["basic_asset_graph"],
+        branch_code_location_to_definitions={
+            "basic_asset_graph": "branch_deployment_change_inputs"
+        },
+    )
+
+    assert differ.get_changes_for_asset(AssetKey("downstream")) == [ChangeReason.INPUTS]
+    assert len(differ.get_changes_for_asset(AssetKey("upstream"))) == 0
+
+
+def test_multiple_changes_for_one_asset(instance):
+    differ = get_parent_asset_graph_differ(
+        instance=instance,
+        code_location_to_diff="code_versions_asset_graph",
+        parent_code_locations=["code_versions_asset_graph"],
+        branch_code_location_to_definitions={
+            "code_versions_asset_graph": "branch_deployment_multiple_changes"
+        },
+    )
+
+    assert differ.get_changes_for_asset(AssetKey("downstream")) == [
+        ChangeReason.CODE_VERSION,
+        ChangeReason.INPUTS,
+    ]
+    assert len(differ.get_changes_for_asset(AssetKey("upstream"))) == 0
+
+
+def test_change_then_revert(instance):
+    differ = get_parent_asset_graph_differ(
+        instance=instance,
+        code_location_to_diff="code_versions_asset_graph",
+        parent_code_locations=["code_versions_asset_graph"],
+        branch_code_location_to_definitions={
+            "code_versions_asset_graph": "branch_deployment_update_code_version"
+        },
+    )
+
+    assert differ.get_changes_for_asset(AssetKey("upstream")) == [ChangeReason.CODE_VERSION]
+    assert len(differ.get_changes_for_asset(AssetKey("downstream"))) == 0
+
+    differ = get_parent_asset_graph_differ(
+        instance=instance,
+        code_location_to_diff="code_versions_asset_graph",
+        parent_code_locations=["code_versions_asset_graph"],
+        branch_code_location_to_definitions={"code_versions_asset_graph": "parent_asset_graph"},
+    )
+
+    assert len(differ.get_changes_for_asset(AssetKey("upstream"))) == 0
+    assert len(differ.get_changes_for_asset(AssetKey("downstream"))) == 0
+
+
+def test_large_asset_graph(instance):
+    differ = get_parent_asset_graph_differ(
+        instance=instance,
+        code_location_to_diff="huge_asset_graph",
+        parent_code_locations=["huge_asset_graph"],
+        branch_code_location_to_definitions={
+            "huge_asset_graph": "branch_deployment_restructure_graph"
+        },
+    )
+
+    for i in range(6, 1000):
+        key = AssetKey(f"asset_{i}")
+        assert differ.get_changes_for_asset(key) == [ChangeReason.INPUTS]
+
+    for i in range(6):
+        key = AssetKey(f"asset_{i}")
+        assert len(differ.get_changes_for_asset(key)) == 0
+
+
+def test_multiple_code_locations(instance):
+    # There are duplicate asset keys in the asset graphs of basic_asset_graph and code_versions_asset_graph
+    # this test ensures that the ParentAssetGraph differ constructs AssetGraphs of the intended code location and does not
+    # include assets from other code locations
```

Comment:
do we not reject conflicts like this somewhere? what happens if you try to load these in to `dagster dev` ?

Reply:
i ran some tests like this in dagster dev and found that i could add the same Definitions object as multiple different code locations with no issues. I could also make a new Definitions object with the same asset keys as another Definitions object and load both of those as separate code locations with no issue